File: common/src/main/java/net/jr/common/Position.java
package net.jr.common;

public class Position {

    private int line;

    private int column;

    public Position(int line, int column) {
        this.line = line;
        this.column = column;
    }

    public int getLine() {
        return line;
    }

    public int getColumn() {
        return column;
    }

    public Position nextLine() {
        return new Position(line+1, 1);
    }

    public Position nextColumn() {
        return new Position(line, column+1);
    }

    @Override
    public String toString() {
        return getLine() + ":" + getColumn();
    }

    @Override
    public int hashCode() {
        return line ^ 7 + column ^ 23;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == null || !obj.getClass().equals(Position.class)) {
            return false;
        }
        final Position oPosition = (Position) obj;
        return oPosition.line == line && oPosition.column == column;
    }
}

File: src/common/Position.ts

export class Position {

    private _line: number;

    private _col: number;

    static start(): Position {
        let p: Position = new Position;
        p._line = 1;
        p._col = 1;
        return p;
    }

    updated(s: string): Position {
        let p: Position = new Position;
        p._line = this._line;
        p._col = this._col;
        for (let c of s) {
            if (c == '\n') {
                p._line = p._line + 1;
                p._col = 1;
            } else {
                p._col = this._col + 1;
            }
        }
        return p;
    }

    get column(): number {
        return this._col;
    }

    get line(): number {
        return this._line;
    }
}